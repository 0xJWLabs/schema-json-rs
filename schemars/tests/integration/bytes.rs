use crate::prelude::*;
use bytes1::{Bytes, BytesMut};

#[test]
fn bytes() {
    test!(Bytes)
        .assert_snapshot()
        .assert_allows_serde_roundtrip([Bytes::new(), Bytes::from_iter([12; 34])])
        .assert_matches_deserialize(arbitrary_values());
}

#[test]
fn bytes_mut() {
    test!(BytesMut)
        .assert_identical::<Bytes>()
        .assert_allows_serde_roundtrip([BytesMut::new(), BytesMut::from_iter([12; 34])])
        .assert_matches_deserialize(arbitrary_values());
}
