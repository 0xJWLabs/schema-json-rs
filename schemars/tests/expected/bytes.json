{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "Tuple_of_Bytes_and_Bytes",
  "type": "array",
  "prefixItems": [
    {
      "$ref": "#/$defs/Bytes"
    },
    {
      "$ref": "#/$defs/Bytes"
    }
  ],
  "minItems": 2,
  "maxItems": 2,
  "$defs": {
    "Bytes": {
      "type": [
        "array",
        "string"
      ],
      "items": {
        "type": "integer",
        "format": "uint8",
        "minimum": 0
      }
    }
  }
}